src/state.cpp

```diff
@@ -26,6 +26,15 @@ namespace hpx
             }
             return (rt->get_thread_manager().status() == st);
         }
+        bool threadmanager_is_at_least(state st)
+        {
+            hpx::runtime* rt = get_runtime_ptr();
+            if (nullptr == rt) {
+                // we're probably either starting or stopping
+                return false;
+            }
+            return (rt->get_thread_manager().status() >= st);
```

Comment:
I hope that the states always form a progression tha increases with some understood metric of what a threadmanager should be doing.